java-checks/src/test/files/checks/security/CryptographicKeySizeCheck.java

```diff
@@ -0,0 +1,130 @@
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import javax.crypto.KeyGenerator;
+
+class A {
```

Comment:
nitpick: there's lots of tests here. I think that grouping the tests in 3 methods would increase readability 

- `public void nonCompliant()`
- `public void falseNegatives()`
- `public void compliant()`

Also, why do you need `class A` and `class B` ?

Reply:
for the false negatives, you can add a comment with the number of the ticket PY is working on (SONARJAVA-2491) and mention it can solve the problem

Reply:
In the test files we don't use names for variables, functions and classes that would provide information to a reader.
From the comments: ` Compliant ` and ` Noncompliant ` someone can easily understand which test cases are supposed to raise an issue or not.

Reply:
I know what the Noncompliant, FN and Compliant are for. What is hard for me is to understand the overall picture, what's the difference between each test, what each test does differently. There are 21 tests here. Are there 21 different situations that this rule is tested for? If yes, I cannot see clearly what these differences are.